src/util/__tests__/createServerRootMixin.test.js

```diff
@@ -0,0 +1,467 @@
+            searchClient: createFakeClient(),
+            indexName: 'hello',
+          }),
+        ],
+        render(h) {
+          return h(InstantSearchSsr, {}, [
+            h(Configure, {
+              attrs: {
+                hitsPerPage: 100,
+              },
+            }),
+            h(SearchBox),
+          ]);
+        },
+        // in test, beforeCreated doesn't have $data yet, but IRL it does
+        created() {
+          this.instantsearch.hydrate({
+            __identifier: 'stringified',
+            hello: serialized,
+          });
+        },
+      };
+
+      const {
+        vm: { instantsearch },
+      } = mount(app);
+
+      expect(instantsearch.__initialSearchResults).toEqual(
+        expect.objectContaining({ hello: expect.any(SearchResults) })
+      );
+
+      expect(instantsearch.__initialSearchResults.hello).toEqual(
+        expect.objectContaining(serialized)
+      );
+    });
+
+    it('accepts non-stringified results', () => {
+      const serialized = createSerializedState();
+      const nonSerialized = new SearchResults(
+        new SearchParameters(serialized._state),
+        serialized._rawResults
+      );
+
+      const app = {
+        mixins: [
+          createServerRootMixin({
+            searchClient: createFakeClient(),
+            indexName: 'movies',
+          }),
+        ],
+        render(h) {
+          return h(InstantSearchSsr, {}, [
+            h(Configure, {
+              attrs: {
+                hitsPerPage: 100,
+              },
+            }),
+            h(SearchBox),
+          ]);
+        },
+        // in test, beforeCreated doesn't have $data yet, but IRL it does
+        created() {
+          this.instantsearch.hydrate({
+            movies: nonSerialized,
+          });
+        },
+      };
+
+      const {
+        vm: { instantsearch },
+      } = mount(app);
+
+      expect(instantsearch.__initialSearchResults).toEqual(
+        expect.objectContaining({ movies: expect.any(SearchResults) })
+      );
+
+      expect(instantsearch.__initialSearchResults.movies).toBe(nonSerialized);
+    });
+
+    it('inits the main index', () => {
+      const serialized = createSerializedState();
+
+      const app = {
+        mixins: [
+          createServerRootMixin({
+            searchClient: createFakeClient(),
+            indexName: 'hello',
+          }),
+        ],
+        render(h) {
+          return h(InstantSearchSsr, {}, [
+            h(Configure, {
+              attrs: {
+                hitsPerPage: 100,
+              },
+            }),
+            h(SearchBox),
+          ]);
+        },
+      };
+
+      const {
+        vm: { instantsearch },
+      } = mount(app);
+
+      expect(instantsearch.mainIndex.getHelper()).toBe(null);
+
+      instantsearch.hydrate({
+        __identifier: 'stringified',
+        hello: serialized,
+      });
+
+      expect(instantsearch.mainIndex.getHelper().constructor.name).toBe(
```

Comment:
Testing `constructor.name` is an implementation detail and can break once we swap the helper implementation. What about asserting with `expect.any(AlgoliaSearchHelper)` as below?

Reply:
It didn’t work for some reason, one test it worked and the next it failed. I think maybe Babel might have been involved in it, but this was the simplest way I found to have a passing test which asserts something

Reply:
Ok, then asserting for not `null` might be enough here?